### PR TITLE
adding tilt-dev/tilt recipe

### PR DESCRIPTION
**Pull request by joseph-zhong**
Hi, this is my first time submitting a recipe PR! Thanks to everyone, the process was relatively smooth getting here.

One confusing part in the documentation is the expected behavior for using the newer MacOS SDK: in https://conda-forge.org/docs/maintainer/knowledge_base.html#requiring-newer-macos-sdks it specifies to edit the `recipe/conda_build_config.yaml`, which in this case I'm assuming should be `tilt/conda_build_config.yaml`, and the former is after the PR is merged and the recipe is uploaded. I've tested the result of rerendering by locally running 

```
conda-smithy rerender
```

and verifies the generated `.ci_support/os64.yaml` generates as expected

@conda-forge-admin, please ping team

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by isuruf** [comment]
Can you build from source?

**Comment by joseph-zhong** [comment]
Hey @isuruf thanks for taking a look -- Are there particular advantages from that as opposed to using the release binaries? It's likely possible, but the current setup is just emulating the recommended installation currently from https://github.com/tilt-dev/tilt/ 

@nicks do you have an opinion here?

**Comment by isuruf** [comment]
Yes, because we don't where the binaries come from. Particularly it looks it is using some go modules and is violating the licenses of those go modules because the pre built binary is not distributing the copyright notices and license texts of its dependencies.

**Comment by joseph-zhong** [comment]
@isuruf Got it -- that makes sense to me, I can look into compiling from source and get back to you on this, thanks again for the quick feedback

**Comment by nicks** [comment]
sgtm, i agree compiling from scratch is more the norm here. 

@joseph-zhong Tilt uses goreleaser to compile the binaries, so the flags you need will look like this:
VERSION=... DATE=... COMMIT=... && go install -tags=osusergo -mod=vendor -ldflags="-s -w -X main.version=0.15.1 -X main.commit=$COMMIT -X main.date=$DATE" ./cmd/tilt

**Comment by joseph-zhong** [comment]
Thanks @nicks -- I was able to compile from source locally, however it doesn't look like go v1.14 is available yet and I'm failing with the following dependency error, it appears we may have to wait for https://github.com/conda-forge/go-feedstock/pull/87

@isuruf Do you have any suggestions otherwise if a dependency is not ready in conda-forge? Is there a way to install go 1.14 from the official source? It appears that's the only build requirement for compilation for both UNIX/Windows.

```
$ conda-build tilt
  File "C:\Users\josephz\Miniconda3\lib\site-packages\conda\resolve.py", line 288, in verify_specs
    raise ResolvePackageNotFound(bad_deps)
conda.exceptions.ResolvePackageNotFound:
  - go_win-64=1.14
```

**Comment by nicks** [comment]
@joseph-zhong can you try building the latest version of Tilt with go v1.13? I tried it out and I think it should work now.

**Comment by joseph-zhong** [comment]
Awesome @nicks! I'll try again and update the PR

**Comment by joseph-zhong** [comment]
@conda-forge-admin please rerender

**Comment by joseph-zhong** [comment]
@conda-forge-admin, please rerender

File: .ci_support/osx64.yaml [hunk 1]
```diff
@@ -1,8 +1,12 @@
-c_compiler:
-  - clang
-cxx_compiler:
-  - clangxx
-fortran_compiler:
-  - gfortran
+MACOSX_DEPLOYMENT_TARGET:
```
Comment (chrisburr) [inline .ci_support/osx64.yaml:1]: Can you revert the changes to this file (after addressing any other comments so we still see it's green). The CI will fail but I'll merge regardless (the `recipes/tilt/conda_build_config.yaml` will make sure the feedstock is correct).

File: recipes/tilt/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+source:
+  url: https://github.com/tilt-dev/tilt/archive/v{{ version }}.tar.gz
+  sha256: 4df87f9efaa83e518d45f0ea10b6b3f7839076ab70731e434ce2f45f794770d4
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('go') }} 1.13
```
Comment (chrisburr) [inline recipes/tilt/meta.yaml:18]: Do you need to specify a version? It should choose the newest by default:
```suggestion
    - {{ compiler('go') }}
```

File: recipes/tilt/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - clangxx_osx-64  # [osx]
```
Comment (chrisburr) [inline recipes/tilt/meta.yaml:19]: Maybe add a comment about why it's only needed for macOS?
```suggestion
    - {{ compiler('c++') }}  # [osx]
```

File: recipes/tilt/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx]
+
+test:
+  commands:
+    - tilt version
+
+about:
+  home: https://github.com/tilt-dev/tilt
+  license: Apache-2.0
+  license_family: Apache
```
Comment (chrisburr) [inline recipes/tilt/meta.yaml:30]: No need for this any more as it's added automatically based on the value of `license`:
```suggestion
```

File: recipes/tilt/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  license_file: LICENSE
```
Comment (chrisburr) [inline recipes/tilt/meta.yaml:31]: See https://github.com/conda-forge/conda-forge.github.io/issues/1052 in particular for go: https://github.com/conda-forge/conda-forge.github.io/issues/1052#issuecomment-639592614

File: recipes/tilt/meta.yaml (hunk 3 above)
Comment (mariusvniekerk) [reply recipes/tilt/meta.yaml:19]: You could also just use the `{{ compiler('cgo') }}`

**Comment by chrisburr** [comment]
It should have been `cxx` instead of `c++`

**Comment by joseph-zhong** [comment]
Thanks @mariusvniekerk and @chrisburr for the comments! I've made sure the build is green for all OSes and made sure to run `go-licenses` to include the licenses for the dependency packages, please let me know if there are any other changes needed!

**Pull request merged**